README.md

```diff
@@ -26,79 +26,172 @@ pip uninstall securedrop-sdk
-host: 127.0.0.1
-scheme: http
-port: 8081
-target_vm: sd-svs
-dev: False
+Can't overwrite existing cassette ('<path-to-cassette-for-a-functional-test>') in your current record mode ('once').
 ```
 
-Then we can create our second developent VM called `sd-svs`, in which we can checkout/develop
-the `securedrop-sdk` project. The required configuration file is at `/etc/sd-sdk.conf`
+The steps to generate new cassettes are split into two sections: [Generating cassettes for API calls over HTTP](#generating-cassettes-for-api-calls-over-http) and [Generating cassettes for API calls over qrexec](#generating-cassettes-for-api-calls-over-qrexec).
+
+### Generating cassettes for API calls over HTTP
+
+1. Start the server in a docker container by running:
+
+    ```bash
+    NUM_SOURCES=5 make dev
+    ```
+
+2. [Skip if adding a new test] Delete the cassettes you wish to regenerate or just delete all yaml files by running:
+
+    ```bash
+    rm data/*.yml
+    ```
+
+3. Generate new cassettes that make API calls over HTTP by running:
+
+    ```bash
+    make test TESTS=tests/test_api.py
+    ```
+Note: Some tests alter source and conversation data on the server so you may need to restart the server in between test runs.
+
+### Generating cassettes for API calls over qrexec
+
+In order to generate cassettes that make API calls over qrexec, you'll need to run the server and proxy on a separate VM by following the steps outlined in the [Set-up for qrexec communication](#set-up-for-qrexec-communication) section. This is so the VM running the tests can communicate over qrexec with the server.
+
+Once your VMs are setup, follow these steps:
+
+1. Start the server in a docker container on a separate VM by running:
+
+    ```bash
+    NUM_SOURCES=5 make dev
+    ```
+
+2. [Skip if adding a new test] Delete the cassettes you wish to regenerate or just delete all json files by running:
+
+    ```bash
+    rm data/*.json
+    ```
+
+3. Comment out the `@dastollervey_datasaver` decorator above the test you want to generate a new cassette for or just generate all new cassettes by commenting out the decorator above all methods in the `test_apiproxy.py::TestAPIProxy` class.
+
+4. Make qrexec calls to the server and collect real response data:
+
+    ```bash
+    make test TESTS=tests/test_apiproxy.py
+    ```
+
+5. Uncomment the `@dastollervey_datasaver` decorator wherever you commented it out.
+6. Record new cassettes from the response data collected in step 4:
+
+    ```bash
+    make test TESTS=tests/test_apiproxy.py
+    ```
+**Note:** If you get a 403 error it's becuase the test is trying to reuse an old TOTP code, so wait for 60 seconds and try again. Some tests alter source and conversation data on the server so you should restart the server in between test runs.
+
+**Note:** Remember that file download checks don't read actual file path in the `APIProxy` tests as it requires QubesOS setup. You can manually uncomment those lines to execute them on QubesOS setup.
```

Comment:
@kushaldas: I'm not sure what you're saying here. The tests that check whether or not a file downloads successfully from the server will fail when regenerating a cassette because it doesn't check the actual file path? I think I hit this but restarting the server and rerunning the tests seemed to make it work. Would appreciate if you could clarify or perhaps we could modify the file-download tests to pass the first time we generate new cassettes?

Reply:
We will have to add the steps to modify `/etc/qubes-rpc/qubes.Filecopy` file to add the following so that the proxy can send in the files properly to the sdk.

```
dev-server dev-client allow
```

Reply:
i see now what you're saying. good catch!